lms/djangoapps/instructor_task/tasks_helper/module_state.py

```diff
@@ -221,6 +225,75 @@ def rescore_problem_module_state(xmodule_instance_args, module_descriptor, stude
 
 
 @outer_atomic
+def override_problem_score_module_state(xmodule_instance_args, module_descriptor, student_module, task_input):
+    '''
+    '''
+    # unpack the StudentModule:
+    course_id = student_module.course_id
+    student = student_module.student
+    usage_key = student_module.module_state_key
+
+    with modulestore().bulk_operations(course_id):
+        course = get_course_by_id(course_id)
+        # TODO: Here is a call site where we could pass in a loaded course.  I
+        # think we certainly need it since grading is happening here, and field
+        # overrides would be important in handling that correctly
+        instance = _get_module_instance_for_task(
+            course_id,
+            student,
+            module_descriptor,
+            xmodule_instance_args,
+            grade_bucket_type='rescore',
+            course=course
+        )
+
+        if instance is None:
+            # Either permissions just changed, or someone is trying to be clever
+            # and load something they shouldn't have access to.
+            msg = "No module {loc} for student {student}--access denied?".format(
+                loc=usage_key,
+                student=student
+            )
+            TASK_LOG.warning(msg)
+            return UPDATE_STATUS_FAILED
+
+        if not hasattr(instance, 'set_score'):
+            msg = "Score override is only valid for scorable components."
+            raise UpdateProblemModuleStateError(msg)
+
+        weighted_override_score = int(task_input['score'])
```

Comment:
As of my latest commit, it will be caught at the API level in a function that checks the arguments for overriding. I should probably give it a specific error message, though.